modules/claims_api/app/models/claims_api/auto_established_claim.rb

```diff
@@ -384,6 +385,38 @@ def transform_service_branch
       form_data['serviceInformation']
     end
 
+    # Legacy claimsApi code previously allowed servicePay-related service branch
+    # values (servicePay.militaryRetiredPay.payment.serviceBranch, servicePay.separationPay.payment.serviceBranch)
+    # that are not accepted by EVSS.
+    # Rather than refuse those invalid values, this maps them to an equivalent value that EVSS will accept.
+    # Note transform_service_branch above only handles cases found in 'serviceInformation.servicePeriod'.
+    def transform_service_pay_service_branch # rubocop:disable Metrics/MethodLength
+      if form_data['servicePay'].present?
+        if form_data['servicePay']['militaryRetiredPay'].present?
+          received_retired_payment = form_data['servicePay']['militaryRetiredPay']['payment']
+          retired_branch = received_retired_payment['serviceBranch']
+          ClaimsApi::Logger.log('526',
+                                detail: "militaryRetiredPay 'serviceBranch' value received is :: #{retired_branch}")
+          received_retired_payment['serviceBranch'] = ClaimsApi::ServiceBranchMapper.new(retired_branch).value
+
+          form_data['servicePay']['militaryRetiredPay']['payment'] = received_retired_payment
+          form_data['servicePay']['militaryRetiredPay']
+          form_data['servicePay']
+        end
+        if form_data['servicePay']['separationPay'].present?
+          received_separation_payment = form_data['servicePay']['separationPay']['payment']
+          separation_branch = received_separation_payment['serviceBranch']
+          ClaimsApi::Logger.log('526',
+                                detail: "separationPay 'serviceBranch' value received is :: #{separation_branch}")
+          received_separation_payment['serviceBranch'] = ClaimsApi::ServiceBranchMapper.new(separation_branch).value
+
+          form_data['servicePay']['separationPay']['payment']
+          form_data['servicePay']['separationPay']
+          form_data['servicePay']
+        end
```

Comment:
correct, those could be removed. you can set the values here in the `if`'s but I would say stepping outside of the `if` and returning `form_data['servicePay']` from the method as a whole would be better.